cupy/core/reduction.pxi

```diff
@@ -449,9 +452,68 @@ class ReductionKernel(object):
 
         in_indexer = Indexer(in_shape)
         out_indexer = Indexer(out_shape)
+        optimize_context = _optimize._get_current_context()
+        if optimize_context is None:
+            block_size = 512
+        else:
+            params = optimize_context.get_params()
+            if params is None:
+                block_size = self._optimize_params(in_args, out_args, in_indexer, out_indexer, contiguous_size, types, stream)
+                optimize_context.set_params(block_size)
+            else:
+                block_size = params
+
         block_size, block_stride, out_block_num = _get_block_specs(
-            in_indexer, out_indexer, contiguous_size)
+            in_indexer, out_indexer, contiguous_size, block_size)
```

Comment:
How about optimizing also `block_stride` and `out_block_num`?

Reply:
Done.
The benchmark further improved to around 1800 us.